pywemo/ssdp.py

```diff
@@ -164,47 +162,42 @@ def location(self):
 
     @property
     def usn(self):
-        """Return usn value."""
+        """Return unique service name."""
         return self.values.get('usn')
 
+    @property
+    def udn(self):
+        """Return unique device name."""
+        usn = self.usn or ''
+        return usn.split('::')[0]
+
     @property
     def description(self):
         """Return the description from the uPnP entry."""
         url = self.values.get('location', '_NO_LOCATION')
+        try:
+            for _ in range(3):
+                try:
+                    xml = requests.get(url, timeout=REQUESTS_TIMEOUT).content
 
-        if url not in UPNPEntry.DESCRIPTION_CACHE:
-            try:
-                for _ in range(3):
-                    try:
-                        xml = requests.get(
-                            url, timeout=REQUESTS_TIMEOUT
-                        ).content
+                    if xml is not None:
+                        tree = et.fromstring(xml)
+                        if tree is not None:
```

Comment:
Can `fromstring` even return None?  I think it returns a valid object or an exception in all cases

Reply:
Good point. We don't need to check the output of `requests.get` either. If it gets an empty response then lxml will also fail and the code will handle the exception below.

```
>>> et.fromstring('')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "src/lxml/etree.pyx", line 3237, in lxml.etree.fromstring
  File "src/lxml/parser.pxi", line 1896, in lxml.etree._parseMemoryDocument
  File "src/lxml/parser.pxi", line 1777, in lxml.etree._parseDoc
  File "src/lxml/parser.pxi", line 1082, in lxml.etree._BaseParser._parseUnicodeDoc
  File "src/lxml/parser.pxi", line 615, in lxml.etree._ParserContext._handleParseResultDoc
  File "src/lxml/parser.pxi", line 725, in lxml.etree._handleParseResult
  File "src/lxml/parser.pxi", line 654, in lxml.etree._raiseParseError
  File "<string>", line 1
lxml.etree.XMLSyntaxError: Document is empty, line 1, column 1
```

Reply:
Right, as long as `get().contents` cannot actually return `None`, since `None` raises a `ValueError` if passed to `fromstring`.  That said, I'm fairly certain `contents` cannot be None

Reply:
Ah bummer. Looks like there is an edge case where `content` can be `None`.

https://github.com/psf/requests/blob/8c211a96cdbe9fe320d63d9e1ae15c5c07e179f8/requests/models.py#L828-L831

Fixed.

Reply:
Always good to check the source - I guess I've just never *seen* it be None!

Reply:
But great fix for that edge case, no need for the `if` statement